Adds a Blob server for getting teh blobs

pietgeursen opened the pull request:
Adds a really basic http server for getting blobs

It does actually work on the app, I've seen images in posts now which is :cake: :tada: :balloon: 

@cryptix at some point we talked about being able to make an async version of the `blobstore.Get` function. I can't remember if it was hard or not. If it's easy, how to do it?

pietgeursen committed:
Creates a http server for getting blobs.

pietgeursen committed:
fmt

cryptix commented:
async in which sense? They it waits until the blob is received?

There is a event-emitter like system on the blobstore that can be used for this but it needs to be wired up by hand right now.

Do you use the http sever just to inject the blobs.want call?

cryptix committed:
don't listenAndServe in the goroutine

this way the error is swallowed if the listen fails

cryptix committed:
add waiting to blobs server

using BlobsGet() bufferes a lot of data on the go sidem, which is
supposed to be passed across the language barrier.

Internally we don't want to copy all that into RAM, this uses the reader
from blobs.Get and copys it into the http response directly without
buffering.

The blobstore.Changes() observable is hooked into to wait if the blob
isn't available yet.

cryptix committed:
fix imports and update vendor

cryptix commented:
i made some polish and also fixed the imports since you did the github move (did not update the deps though and also commited a patch with the diffs from upstream on go-ssb)

this _should_ work and wait for the blob, for 30 seconds at least.

cryptix commented on bot.go:
```diff
@@ -290,14 +292,66 @@ func Start(repoPath string) {
+					return
+				}
 			}
+
```
might want to add some Content-Length or mime types here but it should still work.

cryptix committed:
update dependencies

cryptix committed:
fix start stop

need to close the listener if we want to restart the bot

cryptix commented:
a92fb4 takes care of https://github.com/cryptoscope/ssb/issues/30 and fixes the formatting of the error that came up in https://github.com/sunrise-choir/sunrise-social-android-app/issues/2#issuecomment-567870729

27651c8 makes sure the listener for the http server is freed in `botStop()` so that the bot can restart.
